scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/macros/Macros.scala

```diff
@@ -38,17 +36,14 @@ object Macros {
   def caseClassParquetSchema[T]: String = macro ParquetSchemaProvider.toParquetSchemaImpl[T]
 
   /**
-   * Macro used to generate parquet tuple converter for a given case class.
-   *
-   * @tparam T Case class type that contains primitive or collection type fields or nested case class.
-   * @return Generated parquet converter
+   * Macro generated case class read support
    */
-  def caseClassParquetTupleConverter[T]: ParquetTupleConverter[T] = macro ParquetTupleConverterProvider.toParquetTupleConverterImpl[T]
+  def caseClassParquetReadSupport[T](schema: String): ParquetReadSupport[T] = macro ParquetReadSupportProvider.toParquetReadSupportImpl[T]
```

Comment:
Fixed. At the beginning, I thought if user wanted to create a source/sink for the same I/O path the the schema generation macro will be called twice...so I put it as a parameter so that source/sink could share the same schema instance. But you are right  it requires extra work for the user and I delete it to make it simple to use.